freqtrade/tests/exchange/test_exchange.py

```diff
@@ -1171,3 +1171,46 @@ def test_get_fee(default_conf, mocker):
 
     ccxt_exceptionhandlers(mocker, default_conf, api_mock,
                            'get_fee', 'calculate_fee')
+
+
+def test_stoploss_limit_available_only_for_binance(default_conf, mocker):
+    api_mock = MagicMock()
+    exchange = get_patched_exchange(mocker, default_conf, api_mock)
+    with pytest.raises(NotImplementedError):
+        exchange.stoploss_limit('BTC/ETH', 1, 0.8, 0.79)
+
+
+def test_stoploss_limit_order(default_conf, mocker):
+    api_mock = MagicMock()
+    order_id = 'test_prod_buy_{}'.format(randint(0, 10 ** 6))
+    order_type = 'stop_loss'
+
+    api_mock.create_order = MagicMock(return_value={
+        'id': order_id,
+        'info': {
+            'foo': 'bar'
+        }
+    })
+
+    default_conf['dry_run'] = False
```

Comment:
Please also add a test for dry_run=True - that test would fail now i think (see above comments).

Reply:
it doesn't fail on this one but on cancel_order. going to add.

Reply:
ignore my comment. added tests for dry-run = true